samples/commons/js/LinkedList.js

```diff
@@ -70,9 +79,57 @@ class LinkedList {
             first.next = null;
         }
         else this.head = null;
+        this.sizeOfList--;
         return first;
     }
 
+    insertAt(value, index) {
+        if (index > 0 && index > this.sizeOfList) {
+            return false;
+        } else {
+            const node = new Node(value);
+            let curr, prev;
+            curr = this.head;
+            if (index === 0) {
+                node.next = this.head;
+                this.head = node;
+            } else {
+                curr = this.head;
+                let initialIndex = 0;
+                while (initialIndex < index) {
+                    initialIndex++;
+                    prev = curr;
+                    curr = curr.next;
+                }
+                node.next = curr;
+                prev.next = node;
+            }
+            this.sizeOfList++;
+        }
+    }
+
+    removeAt(index) {
+        if (index > 0 && index > this.sizeOfList) {
```

Comment:
Same comment with this. We're going for zero-based indexing, right?

Reply:
Doing more testing on robust here.